spdlog/src/formatter/pattern_formatter/mod.rs

```diff
@@ -428,7 +432,7 @@ impl PatternContext {
 /// There are 2 approaches to create your own pattern:
 /// - Define a new type and implements this trait;
 /// - Use the [`pattern`] macro to create a pattern from a template string.
-pub trait Pattern: Send + Sync {
+pub trait Pattern: Send + Sync + DynClone {
```

Comment:
Why does `Pattern` requires `Clone` after the change?

Reply:
`Formatter` requires `Clone` to be implemented since 78e359c2682300b00d002881d0c1f8bc17de4458, so that we can implement `set_formatter` for *combined sink* https://github.com/SpriteOvO/spdlog-rs/blob/0a2489c94e4f77b8ce43e5f4eacc49d8a320959d/spdlog/src/sink/async_sink/async_pool_sink.rs#L86-L92

Before this PR, all patterns are actual types, present in the generic `P` of `PatternFormatter<P>`, also all patterns implemented `Clone`, so we didn't explicitly bound `Clone` for trait `Pattern`.

But with runtime pattern, we now have `Box<dyn Pattern>` which doesn't implement `Clone`, if we `impl Clone for Box<dyn Pattern>`, we have to bound `Clone` on trait `Pattern`, and `DynClone` make it possible with object-safe.